File: src/7/7.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	input := bufio.NewScanner(os.Stdin)
	var sums = make([]int, 141)
	sum := 0
	col := (142 / 2) - 1
	end := col + 1
	sums[col] = 1
	i := 0
	for input.Scan() {
		line := input.Text()
		if i == 0 || i%2 == 1 {
			i++
			continue
		}
		for j := col; j < end; j++ {
			if line[j] == '^' && sums[j] > 0 {
				sum++
				sums[j-1] += sums[j]
				sums[j+1] += sums[j]
				sums[j] = 0
			}
		}
		col -= 1
		end += 1
		i++
	}
	sum2 := 0
	for _, val := range sums {
		sum2 += val
	}
	fmt.Println(sum, sum2)

}

File: src/7/7.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
    terminal: false
});

let lines = []
String.prototype.replaceAt = function (index, replacement) {
    return this.substring(0, index) + replacement + this.substring(index + replacement.length);
}

function pascals(numRows) {
    if (numRows === 0) return [];
    if (numRows === 1) return [[1]];
    let result = [];
    for (let row = 1; row <= numRows; row++) {
        let arr = [];
        for (let col = 0; col < row; col++) {
            if (col === 0 || col === row - 1) {
                arr.push(1);
            } else {
                arr.push((result[row - 2][col - 1] + result[row - 2][col]));
            }
        }
        result.push(arr);
    }
    return result;
}

let sums = []

console.log("Go!")
rl.on('line', (line) => {
    lines.push(line.split(""))
    sums.push([])
})

rl.on('close', () => {
    let sum = 0
    for (let i = 0; i < lines.length - 1; i++) {
        for (let j = 0; j < lines[i].length; j++) {
            const el = lines[i][j];
            if (el === "S") {
                lines[i + 1][j] = "|"
                sums[i + 1][j] = 1
                break
            }
            if (el === "|") {
                if (lines[i + 1][j] === "^") {
                    sum++
                    lines[i + 1][j - 1] = "|"
                    lines[i + 1][j + 1] = "|"
                    sums[i + 1][j - 1] = (sums[i + 1][j - 1] ?? 0) + sums[i][j]
                    sums[i + 1][j + 1] = (sums[i + 1][j + 1] ?? 0) + sums[i][j]
                } else {
                    lines[i + 1][j] = lines[i][j]
                    sums[i + 1][j] = (sums[i + 1][j] ?? 0) + sums[i][j]
                }
            }
        }
    }
    console.log(sum, sums[sums.length - 1].reduce((a,b) => a + b))

    // var fs = require('fs');

    // var file = fs.createWriteStream('array.txt');
    // file.on('error', function (err) { /* error handling */ });
    // sums.forEach(function (v) { file.write(v.join(',') + '\n'); });
    // file.end();
})
